R/spq_group_by.R

```diff
@@ -12,13 +12,38 @@
 #' spq_perform()
 #' }
 spq_group_by = function(.query, ...){
+
   vars = purrr::map_chr(rlang::enquos(...), spq_treat_argument)
   varformula = purrr::map_df(.query$select, get_varformula)
-  .query$select = varformula %>%
-    dplyr::filter(.data$name %in% vars) %>%
-    dplyr::pull(.data$full)
-  .query$group_by = varformula %>%
-    dplyr::filter(.data$name %in% {{ vars }}) %>%
-    dplyr::pull(.data$formula)
+
+  vars_selected = vars[vars %in% varformula[["name"]]]
+  vars_formulaed <- dplyr::filter(
```

Comment:
I think that file might be refactored later, when we better keep track of variables used in formulas.